channels/serializers.py

```diff
@@ -888,7 +888,9 @@ def validate_subscriber_name(self, value):
     def create(self, validated_data):
         api = self.context["channel_api"]
         channel_name = self.context["view"].kwargs["channel_name"]
-        return api.add_subscriber(validated_data["subscriber_name"], channel_name)
+        username = validated_data["subscriber_name"]
+        api.add_subscriber(username, channel_name)
+        return list(Channel.objects.get(name=channel_name).subscribers)[-1]
```

Comment:
I don't think this will work reliably since there is no set ordering on the underlying objects. Can you add some logic to sort by the `ChannelSubscription` timestamp?

Reply:
Good catch, done